src/condor_utils/filename_tools_cpp.cpp

```diff
@@ -145,8 +145,18 @@ int filename_remap_find( const char *input, const char *filename, std::string &o
 	p = buffer;
 
 	while(1) {
+		char *start = p;
 		p = copy_upto(p,name,'=',input_len);
 		if(!p) break;
+
+		char *semi = strchr(name, ';');
+		if (semi != nullptr) {
+			// This could happen if a field is missing an equals, e.g.
+			// name1=usr1;missing_equal;name2=usrl1
+			p = start + 1; // skip this field and restart
```

Comment:
The seems way slower than necessary.  Shouldn't you skip to `semi + 1`?

Reply:
This doesn't work because `semi` points into `name`, and `name` is a truncated copy of the whole string.

Reply:
It still seems way slower than necessary. :)  You could do p = start + (semi - name) or so?